File: android/core/domain/src/main/java/com/enterprise/core/domain/repositories/ChatRepository.kt
package com.enterprise.core.domain.repositories

import com.enterprise.core.domain.entities.DomainResult
import com.enterprise.core.domain.entities.UCPMessage
import kotlinx.coroutines.flow.Flow

/** AI chat repository contract with WebSocket streaming. */
interface ChatRepository {
    suspend fun connect()
    suspend fun disconnect()
    suspend fun sendMessage(text: String): DomainResult<UCPMessage>
    fun messageStream(): Flow<UCPMessage>
    suspend fun getHistory(limit: Int = 50): DomainResult<List<UCPMessage>>
}

File: ios/EnterpriseMobileApp/Sources/Core/Domain/Repositories/ChatRepository.swift
import Foundation

/// Repository protocol for AI chat operations (UCP).
public protocol ChatRepository: Sendable {
    func connect() async throws
    func disconnect() async
    func sendMessage(_ text: String) async -> DomainResult<UCPMessage>
    func messageStream() -> AsyncStream<UCPMessage>
    func getHistory(limit: Int) async -> DomainResult<[UCPMessage]>
}
